src/node/routes/vscode.ts

```diff
@@ -33,27 +34,43 @@ export class CodeServerRouteWrapper {
       })
     }
 
-    const { query } = await req.settings.read()
-    if (query) {
-      // Ew means the workspace was closed so clear the last folder/workspace.
-      if (req.query.ew) {
-        delete query.folder
-        delete query.workspace
-      }
+    let folder = undefined
+    let workspace = undefined
+    const to = self(req)
```

Comment:
We could move these into the lower `if` block since they are not used outside it.